### PR TITLE
chore: fix package link script issues

### LINKED ISSUE #6248: bin/package-unlink unlinks all packages instead of the one that was specified
Prerequisites
* [x] Are you running the latest version?
* [x] Are you able to consistently reproduce the issue?
* [x] Did you search the issue queue for existing issue? [Search issues](https://github.com/reactioncommerce/reaction/issues)

## Issue Description
Running `bin/package-unlink some-plugin` (works for all plugins) seems to unlink *all* plugins and not just `some-plugin`. This is the case regardless of whether the plugin was linked with `bin/package-link` beforehand or not.

## Steps to Reproduce
1. Link a plugin with `bin/package-link some-plugin`, then link another plugin with `bin/package-link other-plugin`.
2. Open a shell on your API container with `docker exec -it reaction_api_1 bash`.
3. In your container's shell, go to `/usr/local/src/app/node_modules/@reactioncommerce`.
4. Still in your container's shell, run `ls -la`. Notice you have two symlinks in place: one for `some-plugin` and another for `other-plugin`.
5. On your host machine, run `bin/package-unlink some-plugin` to unlink `some-plugin`.
6. Back into your container's shell, run `ls-ls`. Notice that both `some-plugin` and `other-plugin` were unlinked.

## Possible Solution
No idea. Is this even on Reaction's side or is it an NPM issue?

### LINKED ISSUE #6250: Linked packages don't get updated with new changes
Prerequisites
* [x] Are you running the latest version?
* [x] Are you able to consistently reproduce the issue?
* [x] Did you search the issue queue for existing issue? [Search issues](https://github.com/reactioncommerce/reaction/issues)

## Issue Description
Linking a package with `bin/package-link` works fine. However, linking this same package a second time, once changes were made locally, doesn't update the one in the container. It doesn't matter whether you unlink the package before re-linking it. The behavior is the same.

## Steps to Reproduce
1. Link a package with `bin/package-link`. For the sake of having something to compare, include a `Logger.info("Hello World")` in a query resolver, for example.
2. Call this query, and notice the `Hello World` being printed in the server logs.
3. Remove the `Logger.info` line from the local copy of the package.
4. Unlink this same package with `bin/package-unlink`, then re-link it with `bin/package-link`.
5. After the API has restarted, call the same query again and notice the `Hello World` being printed, even though the `Logger.info` call was removed.

## Possible Solution
Maybe this is due to some NPM cache or something. There's probably a `--no-cache` flag we can use when linking/unlinking packages.

### PR DESCRIPTION
**Pull request by aldeed**
Resolves #6250
Impact: **minor**
Type: **chore**

- npm install official packages after unlink
- don't unlink unless it is truly linked
- require package namespace for both commands

## Breaking changes
The `package-link` and `package-unlink` commands now need the full package name, including `@reactioncommerce/` org prefix.

## Testing
Try it

### CODE REVIEW

**Comment by aldeed** [comment]
I'd like to also figure out a way to maintain the other links when unlinking one, but I think we'd have to create a file to track the links and re-create them all in the unlink script after unlinking. In other words, essentially write our own version of the NPM RFC proposal.

A bigger concern is that I am seeing that often, after I link a second package, the code from the first package is no longer executing, even though the symlink is still there and the symlinked files have the changes. It's like there's a different version of the package also installed somewhere in the `node_modules` tree, and the `import()` is finding that one instead. So far I haven't figured out what's going on.

**Comment by aldeed** [comment]
My "bigger concern" comment is captured in #6250, and I think I actually fixed it by changing `docker cp` command in this PR. So I believe the only remaining issue is that `unlink` unlinks all packages and then you have to relink any you still want linked. Most likely will not figure out a solution for that in this PR.

**Comment by focusaurus** [comment]
OK I think I encountered some of what is being discussed in #6250 today so I'm going to test this branch and see if it helps. One thing I'd like to confirm about this tooling, is it:

- use BOTH package-link and a docker volume mount

or is this package-link workflow designed to work without an extra docker volume mount for the api-plugin you're working on?

**Comment by kieckhafer** [comment]
> OK I think I encountered some of what is being discussed in #6250 today so I'm going to test this branch and see if it helps. One thing I'd like to confirm about this tooling, is it:
> 
> 
> 
> - use BOTH package-link and a docker volume mount
> 
> 
> 
> or is this package-link workflow designed to work without an extra docker volume mount for the api-plugin you're working on?

When testing originally, I did not have to add the additional volume mount. I followed the example workflow here and was able to link and unlink: https://github.com/reactioncommerce/reaction/pull/6238 I only did a single time, so did not run into the issues of #6250

**Comment by kieckhafer** [comment]
I’m going to hold off on the 3.8.0 release until this is merged. The only big change in that release is the original PR which this current PR is fixing, so makes sense to wait. cc @mdelreal

**Comment by loan-laux** [comment]
I gave it another try with @aldeed's latest changes and didn't run into the issues mentioned in #6250 anymore. Been using `bin/package-link` for the whole day and it seems very stable. On my end, seems like this is good to go.

**Comment by kieckhafer** [comment]
I've also tested this out and it seems to be good. @focusaurus have you had a chance to try it out?

**Review by focusaurus — APPROVED** [review]

**Comment by focusaurus** [comment]
Yeah I've used this a few times. I'm still not 100% clear on the workflow and the various levels of technical challenges and solutions that are in play here, but it seems to work OK for me.

**Pull request merged**
